lib/rust/parser/src/macros/built_in.rs

```diff
@@ -715,14 +715,33 @@ fn freeze<'s>() -> Definition<'s> {
     crate::macro_definition! {("FREEZE", everything()) capture_expressions}
 }
 
+/// private can be either specified as the very first statement in the module, marking the
+/// whole module as private. Or it can be prepended to some definitions. For example it can
+/// be prepended to atom constructor definition.
 fn private_keyword<'s>(
     segments: NonEmptyVec<MatchedSegment<'s>>,
     precedence: &mut operator::Precedence<'s>,
 ) -> syntax::Tree<'s> {
     let segment = segments.pop().0;
     let keyword = into_private(segment.header);
-    let body = precedence.resolve(segment.result.tokens());
-    syntax::Tree::private(keyword, body)
+    let body_opt = precedence.resolve(segment.result.tokens());
+    match body_opt {
+        Some(body) => {
+            let body_ = body.clone();
+            let new_body = to_body_statement(body_);
```

Comment:
> I think this test would fail: `expect_invalid_node("private ConstructorOutsideType")`. Whether to_body_statement is applied must depend on whether the expression is in the top level of a type definition, not the expression's content.

@kazcw Yes, it fails. I have added that test in 395544f9df1464bbac86322f6c9de0a25329feed.

Your proposed solution would work for private constructors. But in the near future, we would like to support private methods as well. Not only methods in types, but also static methods in top scope in a module. Any suggestions how to support the private constructor and be able to parse also private top-scope method in the future?

Reply:
`expression_to_statement` (where method definitions are recognized) would need a clause analogous to the new logic in `to_body_statement`.

Reply:
@kazcw Please help. I am struggling to implement your suggestion in https://github.com/enso-org/enso/pull/9692/commits/fcf2db5b3aff25d9deec300af1ac4770d4ca3a9a. That commit does not compile - there is a borrow checker error that I cannot resolve. Would you please be so kind and fix that commit?

Reply:
I refactored it, pushed @ [4a11104](https://github.com/enso-org/enso/pull/9692/commits/4a11104ef10debd38bd9fe3796f81d01928821f2)